common/coconut-interface/src/lib.rs

```diff
@@ -0,0 +1,170 @@
+pub use coconut_rs::*;
+use coconut_rs::{Attribute, Base58, BlindSignRequest, BlindedSignature, PublicKey};
+use getset::{CopyGetters, Getters};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Getters, CopyGetters, Clone)]
+pub struct Credential {
+    #[getset(get = "pub")]
+    n_params: u32,
+    #[getset(get = "pub")]
+    theta: Theta,
+    public_attributes: Vec<String>,
+    #[getset(get = "pub")]
+    signature: Signature,
+}
+impl Credential {
+    pub fn new(
+        n_params: u32,
+        theta: &Theta,
+        public_attributes: &[Attribute],
+        signature: &Signature,
+    ) -> Credential {
+        Credential {
+            n_params,
+            theta: theta.clone(),
+            public_attributes: public_attributes
+                .iter()
+                .map(|attr| attr.to_bs58())
+                .collect(),
+            signature: *signature,
+        }
+    }
+
+    pub fn public_attributes(&self) -> Vec<Attribute> {
+        self.public_attributes
+            .iter()
+            .map(|x| Attribute::try_from_bs58(x).unwrap())
+            .collect()
+    }
+
+    pub fn verify(&self, validator_urls: Vec<String>) -> bool {
```

Comment:
I think it would make more sense if the public API took verification key as an argument rather than urls of validators